@xen-orchestra/fs/src/s3.js

```diff
@@ -242,15 +288,56 @@ export default class S3Handler extends RemoteHandlerAbstract {
   // s3 doesn't have a rename operation, so copy + delete source
   async _rename(oldPath, newPath) {
     await this.copy(oldPath, newPath)
-    await this._s3.deleteObject(this._createParams(oldPath))
+    await this._s3.send(new DeleteObjectCommand({ ...this._createParams(oldPath) }))
   }
 
   async _getSize(file) {
     if (typeof file !== 'string') {
       file = file.fd
     }
-    const result = await this._s3.headObject(this._createParams(file))
-    return +result.ContentLength
+    try {
+      const result = await this._s3.send(new HeadObjectCommand({ ...this._createParams(file) }))
+
+      return +result.ContentLength
+    } catch (error) {
+      if (error.$response.statusCode === 404) {
+        return 0
```

Comment:
Why this change?

It appears to be a weird behavior in our lib to returns 0 for  missing entry :thinking: 

What's the behavior of a `Local` remote?